go/vt/vttablet/tabletserver/vstreamer/vstreamer_test.go

```diff
@@ -546,7 +546,55 @@ func TestBuffering(t *testing.T) {
 			`type:DDL ddl:"alter table packet_test change val val varchar(128)" `,
 		}},
 	}}
-	runCases(t, nil, testcases)
+	runCases(t, nil, testcases, "")
+}
+
+func TestBestEffortNameInFieldEvent(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	filter := &binlogdatapb.Filter{
+		BestEffortNameInFieldEvent: true,
+		Rules: []*binlogdatapb.Rule{{
+			Match: "/.*/",
+		}},
+	}
+	// Modeled after vttablet endtoend compatibility tests.
+	execStatements(t, []string{
+		"create table vitess_null(id int, val varbinary(128), primary key(id))",
+	})
+	position := masterPosition(t)
+	execStatements(t, []string{
+		"insert into vitess_null values(1, 'abc')",
+		"rename table vitess_null to vitess_null_new",
+	})
+
+	defer execStatements(t, []string{
+		"drop table vitess_null_new",
+	})
+	engine.se.Reload(context.Background())
+	testcases := []testcase{{
+		input: []string{
+			"insert into vitess_null_new values(2, 'abc')",
+		},
+		output: [][]string{{
+			`gtid|begin`,
+			`gtid|begin`,
+			`type:FIELD field_event:<table_name:"vitess_null" fields:<name:"@1" type:INT32 > fields:<name:"@2" type:VARCHAR > > `,
```

Comment:
in this case, we don't have information about vitess_null since it was renamed to vitess_null_test. information returned by binlog for val column == varchar (rather than varbinary). Schema.Engine has correct information.

Reply:
This should be added to the test as comments.